### PR TITLE
OAuth2 Files

### PR DESCRIPTION
**Pull request by anjuthomas**

### CODE REVIEW

File: src/oauth2/init.js [hunk 1]
```diff
@@ -1,34 +1,41 @@
...
+var Service = require('../service');
 var service = require('../oauth2/service');
 
-var DEFAULT_SERVICES = ['Authorization', 'AccessToken',
-                    'RefreshAccessToken', 'ProviderInfoDiscovery'];
+var DEFAULT_SERVICES = [
+  'Authorization', 'AccessToken', 'RefreshAccessToken', 'ProviderInfoDiscovery'
+];
 
 ExpiredToken.prototype = new OicCliError();
```
Comment (TMSCH) [inline src/oauth2/init.js:14]: Could you use a library to use inheritance? Or move the logic to a separate file as it's used in different parts.

File: src/oauth2/init.js [hunk 2]
```diff
@@ -1,34 +1,41 @@
...
 ExpiredToken.prototype = Object.create(OicCliError.prototype);
 ExpiredToken.prototype.constructor = ExpiredToken;
 
-function ExpiredToken(){
-    OicCliError.call();
+function ExpiredToken() {
```
Comment (TMSCH) [inline src/oauth2/init.js:18]: Move constructor above prototype definition.

File: src/oauth2/init.js [hunk 3]
```diff
@@ -1,34 +1,41 @@
...
+  }
+  return service;
 };
 
-function Client(){
-};
+function Client() {};
 
 /**
  * :param ca_certs: Certificates used to verify HTTPS certificates
```
Comment (TMSCH) [inline src/oauth2/init.js:41]: You should use jsdoc for comments.

File: src/oauth2/init.js [hunk 4]
```diff
@@ -38,82 +45,94 @@ function Client(){
...
-    var srvs = services || DEFAULT_SERVICES;
-    this.service = buildServices(srvs, this.serviceFactory, this.http, keyJar, cam);
-    this.clientInfo.service = this.service;
-    this.verifySsl = verifySsl;
+Client.prototype.init = function(
+    clientAuthnMethod, config, caCerts, keyjar, verifySsl, clientCert, httpLib,
+    services, serviceFactory) {
+  caCerts = caCerts || null;
+  clientAuthnMethod = clientAuthnMethod || null;
+  keyjar = keyjar || null;
```
Comment (TMSCH) [inline src/oauth2/init.js:53]: Here it's `keyjar`, below it's `keyJar`. These are not the same variables.

File: src/oauth2/service.js [hunk 5]
```diff
@@ -1,302 +1,325 @@
...
 
-function postXParseResponse(resp, cliInfo, state){
-    state = state || '';
-    if (resp instanceof (AuthorizationResponse, AccessTokenResponse)){
-        cliInfo.stateDb.addMessageInfo(resp, state);
-    }
+function postXParseResponse(resp, cliInfo, state) {
+  state = state || '';
+  cliInfo.stateDb.addResponse(resp, state);
 };
```
Comment (TMSCH) [inline src/oauth2/service.js:15]: Remove trailing ; here and below when declaring functions (not variables or properties).

File: src/oauth2/service.js [hunk 6]
```diff
@@ -1,302 +1,325 @@
...
-  
-Authorization.prototype.init = function(httpLib, keyJar, clientAuthnMethod){
-    httpLib = httpLib || null;
-    keyJar = keyJar || null;
-    clientAuthnMethod = clientAuthnMethod || null;
-    Service.call(httpLib, keyJar, clientAuthnMethod);
-    this.preConstruct.append(this.oauthPreConstruct);
-    this.postParseResponse.append(postXParseResponse);
+
+Authorization.prototype.msgType = oauth2.AuthorizationRequest;
```
Comment (TMSCH) [inline src/oauth2/service.js:37]: Properties should be defined in constructor.

**Review by TMSCH — CHANGES_REQUESTED** [review]
Add tests, there's a lot of new code in this PR.

File: src/oauth2/init.js (hunk 1 above)
Comment (anjuthomas) [reply src/oauth2/init.js]: Fixed

**Pull request merged**